modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java

```diff
@@ -462,11 +462,9 @@ private void _completeOrderWithoutPaymentMethod(
 		long userId = _portal.getUserId(httpServletRequest);
 
 		CommerceOrder commerceOrder =
-			_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
-
-		_commerceOrderLocalService.updatePaymentStatusAndTransactionId(
-			userId, commerceOrderId, CommerceOrderConstants.PAYMENT_STATUS_PAID,
-			StringPool.BLANK);
```

Comment:
@luca-pellizzon not sure if this was wanted to avoid creating a subscription when no payment method is selected. In that case this change is wrong.

Reply:
You should not be able to place a subscription without a "valid" payment method, that means a subscription enabled payment method. This is true for both payment and delivery subscriptions.
There was a check during the checkout that forced to select a valid payment method if a subscription order was checking out. Probably something changed.